arm-service-map/2015-11-01-preview/swagger/arm-service-map.json

```diff
@@ -1299,40 +1215,235 @@
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      },
+      "post": {
+        "tags": [
+          "MachineGroups"
+        ],
+        "operationId": "MachineGroups_Create",
+        "description": "Creates a new machine group.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "machineGroup",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MachineGroup"
+            },
+            "description": "Machine Group resource to create."
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/MachineGroup"
+            }
+          },
+          "default": {
+            "description": "An error occurred while processing the request. See the error.code parameter to identify the specific error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/features/serviceMap/machineGroups/{machineGroupName}": {
+      "get": {
+        "tags": [
+          "MachineGroups"
+        ],
+        "operationId": "MachineGroups_Get",
+        "description": "Returns the specified machine group.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/MachineGroupNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/MachineGroup"
+            }
+          },
+          "default": {
+            "description": "An error occurred while processing the request. See the error.code parameter to identify the specific error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "MachineGroups"
+        ],
+        "operationId": "MachineGroups_Update",
```

Comment:
MachineGroup_Update is only an update. 
MachineGroups have a unique "displayName" within a workspace and the user must be allowed to change it. We initially wanted to use the "name" field of the ARM resource for that purpose, but that field cannot be changed. So we use an "action" to create the MachineGroup, which sets the "name" field to a server-side generated unique value and from there on the created entity can be updated using PUT.